src/node.rs

```diff
@@ -109,3 +99,24 @@ impl Node {
         None
     }
 }
+
+impl TaskWaiting {
+    /// Determine if we are still queued.
+    ///
+    /// Returns `Some` with the entry ID if we are no longer queued.
+    pub(crate) fn status(&self) -> Option<NonZeroUsize> {
+        NonZeroUsize::new(self.entry_id.load(Ordering::Acquire))
+    }
+
+    /// Register a listener.
+    pub(crate) fn register(&self, task: Task) {
+        // Set the task.
+        self.task.store(Some(task));
```

Comment:
I guess potentially overwriting the currently stored task is ok... why?

Reply:
I don't think it is, actually. I wake it up now.